src/HashingLogic.ts

```diff
@@ -0,0 +1,124 @@
+import {AttestationTypeID} from './AttestationTypes'
+
+const {soliditySha3} = require('web3-utils')
+const uuid = require('uuidv4')
+
+export const generateAttestationRequestNonceHash = () =>
+  soliditySha3({type: 'string', value: uuid()})
+
+export interface IAttestationData {
+  /**
+   * The type of attestation (phone, email, etc.)
+   */
+  type: keyof typeof AttestationTypeID
+  /**
+   * Optionally identifies service used to perform attestation
+   */
+  provider?: string
+  // tslint:disable:max-line-length
+  /**
+   * String representation of the attestations data.
+   *
+   * ### Examples ###
+   * email: "test@bloom.co"
+   * sanction-screen: {\"firstName\":\"FIRSTNAME\",\"middleName\":\"MIDDLENAME\",\"lastName\":\"LASTNAME\",\"birthMonth\":1,\"birthDay\":1,\"birthYear\":1900,\"id\":\"a1a1a1a...\"}
+   *
+   * Any attestation that isn't a single string value will be
+   * a JSON string representing the attestation data.
+   */
+  // tslint:enable:max-line-length
+  data: string
+  /**
+   * Attestation type nonce
+   */
+  nonce: string
+  /**
+   * Semantic version used to keep track of attestation versions
+   */
+  version: string
+}
+
+export const hashAttestations = (attestations: IAttestationData[]) => {
+  const individualAttestationHashes = attestations.map(a =>
+    soliditySha3({
```

Comment:
choosing soliditySha3 wasn't a fully educated decision. I just used it because it was available. is there an argument for using a different one? if not, this is fine